airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -595,3 +602,10 @@ def find_properties_object(path: List[str], field: str, properties) -> Dict[str,
                 if child:
                     result.update(child)
     return result
+
+
+def get_table_name(base_table_name: str, suffix: str, lineage_hash: str) -> str:
```

Comment:
how is this dealing with truncation of table names that are too long? There's a lot of discussion in the original issue on this topic: https://github.com/airbytehq/airbyte/issues/2055. i think the case where the base table name is equal to or greater than the allowed length of a table name is a case that we need to handle.

Reply:
It's wrapped with the standard table name truncator that already existed. The test table names are all truncated.

Reply:
Is that truncation scheme something we want to stick with? I feel like there are a lot of cases that it handles poorly. A truncation scheme that is parent / child aware might be stronger?

also how does `TRUNCATE_RESERVED_SIZE` in the existing truncate work with the scheme you're proposing? isn't the hash at the end very similar to what you're doing in your naming scheme?